content/code/testing.md

```diff
@@ -24,8 +31,37 @@ The best way to go about it is to follow the [official guidelines of the Go `tes
 >   func TestT_M() { ... }
 > ```
 
-When it comes to subtests, the names of individual subtests should be lowercased and concise.
-The tests usually start with a subtest called `nominal case` which verifies that the tested component behaves as expected in a baseline situation, where no failures occur and no edge cases are handled.
+You might sometimes want to have multiple test functions for a single method, in which case the tests should be prefixed as defined above, and followed with the purpose of the test:
+
+```go
+func TestMyType_MethodDoesSomethingWhenX(t *testing.T) { ... }
+func TestMyType_MethodDoesHandlesXFailure(t *testing.T) { ... }
+// And so on.
+```
+
+When it comes to subtests, the names of individual subtests should be lowercased and concise. The tests usually start
```

Comment:
WDYM?

Reply:
I mean for example when to have 
```
func TestT_M() { ... }
func TestT_N() { ... }
```

as opposed to this: 

```
func TestT() {
t.Run()...
t.Run()...
}
```

Unless one should always follow one of the above.

Reply:
We would never do

```go
func TestT() {
	t.Run()...
	t.Run()...
}
```

Because that would mean we are testing the type itself. We always test methods, so in the case of your other example, we test two methods, and each method might have subtests, and/or even multiple tests for the same methods:

```go
func TestT_M() {
	t.Run()...
	t.Run()...
}

func TestT_MDoesSomething() {
	// ...
}

func TestT_N() {
	t.Run()...
	t.Run()...
}
```

Reply:
As for when to use one function with tons of subtests to cover the whole method versus using multiple functions, I think there's no clear rule, but usually the point is that ideally we want to have one test function with subtests, and if one specific test case would be too different from the test, or require some very weird logic, we might want to separate it into a different test function. But once again, there's no clear rule for this :)